File: bddtests/src/main/kotlin/worker/models/appointments/AppointmentBookRequest.kt
package worker.models.appointments

import mocking.emis.models.SlotTypeStatus

data class AppointmentBookRequest(
        var userPatientLinkToken: String? = null,
        var slotId: String? = null,
        var bookingReason: String? = null,
        var startTime: String? =null,
        var endTime: String? = null,
        var telephoneNumber: String? = null,
        var telephoneContactType: String? = null,
        var channel: SlotTypeStatus = SlotTypeStatus.Unknown
)

File: backendworker/NHSOnline.Backend.Worker/GpSystems/Appointments/Models/AppointmentBookRequest.cs
using System;
using System.ComponentModel.DataAnnotations;
using NHSOnline.Backend.Worker.ValidationAttributes;

namespace NHSOnline.Backend.Worker.GpSystems.Appointments.Models
{
    public class AppointmentBookRequest
    {
        [Required]
        public string SlotId { get; set; }

        [SafeString]
        [MaxLength(150)]
        public string BookingReason { get; set; }
        public DateTimeOffset? StartTime { get; set; }
        public DateTimeOffset? EndTime { get; set; }
        public string TelephoneNumber { get; set; }
    }
}